src/Fixer/StringNotation/HeredocClosingMarkerFixer.php

```diff
@@ -0,0 +1,172 @@
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\StringNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class HeredocClosingMarkerFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Unify `heredoc` or `nowdoc` closing marker.',
+            [
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD,
+                    ['closing_marker' => 'EOF']
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD,
+                    ['explicit_heredoc_style' => false]
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_START_HEREDOC);
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(
+                'closing_marker',
+                'Preferred closing marker.'
+            ))
+                ->setAllowedTypes(['string'])
+                ->setDefault('EOD')
+                ->getOption(),
+            (new FixerOptionBuilder(
+                'reserved_closing_markers',
+                'Reserved closing markers to be kept unchanged.'
+            ))
+                ->setAllowedTypes(['array'])
+                ->setDefault([
+                    'CSS',
+                    'DIFF',
+                    'HTML',
+                    'JS',
+                    'JSON',
+                    'MD',
+                    'PHP',
+                    'PYTHON',
+                    'RST',
+                    'TS',
+                    'SQL',
+                    'XML',
+                    'YAML',
+                ])
```

Comment:
This list will never be always accurate, while you only give a few examples of markers that should be replaced (`EOT` and `EOF`). Instead of excluding markers, I think it would be easier to give the ones to replace. We could even merge options `reserved_closing_markers` and `closing_marker` so that users can provide their own replacement map, e.g.:
```php
[
    'EOT' => 'EOD',
    'EOF' => 'EOD',
]
```

Reply:
I am not sure... I believe there _is_ some kind of closed set of functional endings, outside of which any other ending should be normalised. The difference in the approach is that with @mvorisek's you will get the diff applied/displayed and you must make decision if you should exclude ending from fixer or accept the change, while in @julienfalque non-listed endings won't be covered until someone discovers they exist in the codebase and adds mapping for them. Personally I prefer the former, because it enforces explicit decision over any found ending marker.

Reply:
Let's keep the "reserved list". In a typical project, there are typically not many, very often zero, additional reserved closing markers needed outside the one predefined.